tests/contrib/test_openai.py

```diff
@@ -0,0 +1,1018 @@
+                    ),
+                    ResponseOutputMessage(
+                        id="",
+                        content=[
+                            ResponseOutputText(
+                                text="Granada",
+                                annotations=[],
+                                type="output_text",
+                            )
+                        ],
+                        role="assistant",
+                        status="completed",
+                        type="message",
+                    ),
+                ],
+                usage=Usage(),
+                response_id=None,
+            )
+        )
+    responses.append(
+        ModelResponse(
+            output=[
+                ResponseOutputMessage(
+                    id="",
+                    content=[
+                        ResponseOutputText(
+                            text='{"follow_up_questions":[], "markdown_report":"report", "short_summary":"rep"}',
+                            annotations=[],
+                            type="output_text",
+                        )
+                    ],
+                    role="assistant",
+                    status="completed",
+                    type="message",
+                )
+            ],
+            usage=Usage(),
+            response_id=None,
+        )
+    )
+
+
+@workflow.defn
+class ResearchWorkflow:
+    @workflow.run
+    async def run(self, query: str):
+        return await ResearchManager().run(query)
+
+
+async def test_research_workflow(client: Client):
+    if sys.version_info < (3, 11):
+        pytest.skip("Open AI support has type errors on 3.9")
+
+    global response_index
+    response_index = 0
+
+    with set_open_ai_agent_temporal_overrides(
+        start_to_close_timeout=timedelta(seconds=10)
+    ):
+        model_activity = ModelActivity(
+            TestProvider(
+                TestResearchModel(  # type: ignore
+                    "", openai_client=AsyncOpenAI(api_key="Fake key")
+                )
+            )
+        )
+        async with new_worker(
+            client,
+            ResearchWorkflow,
+            activities=[model_activity.invoke_model_activity, get_weather],
+        ) as worker:
+            workflow_handle = await client.start_workflow(
+                ResearchWorkflow.run,
+                "Caribbean vacation spots in April, optimizing for surfing, hiking and water sports",
+                id=f"research-workflow-{uuid.uuid4()}",
+                task_queue=worker.task_queue,
+                execution_timeout=timedelta(seconds=10),
+            )
+            result = await workflow_handle.result()
+            activity_count = 0
+            async for e in workflow_handle.fetch_history_events():
+                if e.HasField("activity_task_completed_event_attributes"):
+                    activity_count += 1
+            assert activity_count == 12
+            assert result == "report"
```

Comment:
Probably worth verifying you see some of the result text (from tool/agent calls) you expected to see in the completion events too (applies to other tests as well).